File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//服务层
app.service('brandService', function ($http) {

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../brand/findAllBrand');
    }
    //分页
    this.findPageBrand = function (page, rows) {
        return $http.get('../brand/findPageBrand?page=' + page + '&rows=' + rows);
    }
    //查询实体
    this.findOneBrand = function (id) {
        return $http.get('../brand/findOneBrand?id=' + id);
    }

    this.create = function (entity) {
        return $http.post('../brand/createBrand', entity);
    }

    this.update = function (entity) {
        return $http.post('../brand/updateBrand', entity);
    }

    this.delete = function (ids) {
        return $http.post('../brand/deleteBrand?ids=' + ids);
    }

    this.search = function (page, rows, searchEntity) {
        return $http.post('../brand/search?page=' + page + "&rows=" + rows, searchEntity);
    }
});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 返回所有品牌
     *
     * @return
     */
    List<TbBrand> findAllBrand();

    /**
     * 返回分页列表信息
     *
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPageBrand(int pageNum, int pageSize);

    TbBrand findBrandById(Long id);

    void createBrand(TbBrand brand);

    void updateBrand(TbBrand brand);

    void deleteBrand(Long[] ids);

    PageResult findPage(TbBrand brand, int pageNum, int pageSize);

    /**
     * 品牌下拉框数据
     */
    List<Map> selectOptionList();
}
